editor.planx.uk/src/pages/FlowEditor/lib/store/preview.ts

```diff
@@ -344,6 +346,7 @@ export const previewStore: StateCreator<
         });
       }
     }
+    updateSectionData();
```

Comment:
I'm not massively keen on adding more complexity and logic to `record()`, but it's consistently called on 'forwards', 'back' and 'change' navigation, which is when we want to check our section.